outputs.tf

```diff
@@ -8,13 +8,13 @@ output "nlb_dns_name" {
   value       = aws_lb.main.dns_name
 }
 
-output "nlb_elastic_ips" {
-  description = "List of public Elastic IP addresses associated with the NLB."
+output "nlb_ips" {
+  description = "List of IP addresses associated with the NLB."
 
   value = [
-    data.aws_eip.nlb_eip1.public_ip,
-    data.aws_eip.nlb_eip2.public_ip,
-    data.aws_eip.nlb_eip3.public_ip,
+    local.use_eips ? data.aws_eip.nlb_eip1[0].public_ip : local.use_ipv4_addrs ? var.nlb_ipv4_addrs[0] : "undefined",
```

Comment:
I'm not sure if this is the right approach or if I should do something like this to simplify it. I'm interested in folks opinions on this.
```terraform
local.use_eips ? data.aws_eip.nlb_eip1[0].public_ip : var.nlb_ipv4_addrs[0]
```

Reply:
This seems okay, but I would expect the module to be pretty broken in other ways if both `use_eips` and `use_ipv4_addrs` are false.

Reply:
@dynamike Agreed. I tried to capture that idea in the readme ("If neither option is defined, the module will fail."). Do you think it warrants further error checking or should we just let it fail on its own? Not sure if there is a best practice with that.

Reply:
Error checking in Terraform is fairly limited, you can do validation checking of variables, but not much else. I think what you have now is fine. 

https://www.terraform.io/docs/language/functions/can.html

Reply:
Thanks for the link. I didn't know about `can` or `try`.